pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -204,6 +208,9 @@ if [ "$action" = edit ]; then
 fi
 
 if [ "$action" = switch ] || [ "$action" = build ] || [ "$action" = check ]; then
+  if [ -n "$upgrade" ]; then
+    nix-channel --update
+  fi
```

Comment:
```suggestion
```

Could you backport the code changes from `nixos-rebuild` to keep the logic more in line:

https://github.com/NixOS/nixpkgs/blob/8ee1e50668dc70e118a2df57f4fcadced15c5cbe/pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh#L376-L394